hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java

```diff
@@ -16,93 +16,90 @@
 
 package com.hazelcast.sql.impl.plan.cache;
 
-import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.config.IndexType;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.map.IMap;
-import com.hazelcast.sql.SqlResult;
-import com.hazelcast.sql.SqlRow;
-import com.hazelcast.sql.impl.QueryException;
-import com.hazelcast.sql.impl.SqlResultImpl;
-import com.hazelcast.sql.impl.exec.FaultyExec;
-import com.hazelcast.sql.impl.exec.scan.MapScanExec;
-import com.hazelcast.sql.impl.plan.Plan;
-import com.hazelcast.test.HazelcastParallelClassRunner;
-import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.sql.impl.optimizer.PlanKey;
+import com.hazelcast.sql.impl.optimizer.SqlPlan;
+import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.QuickTest;
-import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hazelcast.sql.impl.SqlTestSupport.nodeEngine;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotSame;
-import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertSame;
-import static org.junit.Assert.fail;
 
-@RunWith(HazelcastParallelClassRunner.class)
```

Comment:
Why it cannot run in parallel anymore?

Reply:
Due to sporadic failures. I tried to make all data independent, but have no luck with that.

Reply:
What kind of failures?

Reply:
Random test methods fails due to wrong result.

Reply:
It extends in the end `SimpleTestInClusterSupport` which should ensure test isolation - bunch of tests already do that and they don't experience any problems. Maybe, the root cause that lie somewhere in tests itself?

Reply:
Long story short, instance has only plan cache is concurrently reused in every test. With serial execution it will be used once per test.

Reply:
But tests are already executed serially, even with `HazelcastParallelClassRunner` - see https://github.com/hazelcast/hazelcast/pull/19162/files#r683214715